exporter/otelarrowexporter/go.mod

```diff
@@ -0,0 +1,90 @@
+module github.com/open-telemetry/opentelemetry-collector-contrib/exporter/otelarrowexporter
```

Comment:
why is it called otelarrowexporter? Why not arrowexporter?

Reply:
These names identify the project, "OTel Arrow", as in the donation. https://github.com/open-telemetry/community/issues/1332

Not sure if this is enough of an answer. I think probably there are other uses for Arrow in OTel in the sense that we also want a Parquet file exporter. The otelarrow exporter/receiver are meant for building a bridge between OTel collectors (via Arrow), whereas the Parquet file exporter (which can also based on OTel-Arrow translation) might be named `otelparquet`. How does this sound to you?

Reply:
Naming is hard. A few observations from the peanut gallery:
* Otel is not a trademark nor does it bear any significance to the casual user. You might be creating yourself some work to explain what this is about. I know Otel is used colloquially to refer to OpenTelemetry, but I'm not aware that we're serious about owning that name.
* `Otel Arrow` seems to then designate a solution based for OpenTelemetry. I am not aware of alternates. Is there an other arrow solution I should be aware of? Are you trying to distinguish this approach against something else I'm not aware of? Is the name not redundant?

Reply:
We are saying that "OTel Arrow" is the name of the protocol, and you are asking whether we should use something shorter, less colloquial and/or longer, I take it.

In my opinion, we chose "OTel Arrow" to designate a solution for OpenTelemetry, because Arrow doesn't define the way to represent OTLP streams, OTel Arrow does. It is redundant in the sense that the "O" in OTLP is redundant, and we chose `otelarrow` to replace `otlp` in the derivative components because to be short and memorable. We already have redundancy in the URL--I don't think we need a third "open telemetry" in e.g., `github.com/open-telemetry/opentelemetry-collector-contrib/exporter/opentelemetryarrowexporter`. 

That said, thank you @atoulme this is a question worth asking, and one we swept aside as the donation was being processed. I'm open to naming this "arrowexporter" and "arrowreceiver". 

@lquerel I'd like your thoughts.
@tigrannajaryan I'd like your thoughts.

Reply:
Please direct feedback to #26491.

Reply:
As the agreed-upon protocol name is "OTel Arrow Protocol" — to emphasize a clear distinction from OTLP while still indicating affiliation with the OpenTelemetry project and integration with Apache Arrow — I believe it is logical to use `otelarrowexporter` and `otelarrowreceiver` in order to stay consistent with the initial reasoning. This approach underscores the universal character of the approach (as opposed to a contribution specific to one vendor) and maintains clarity, even at the cost of minor redundancy. Obviously if the preference is to have a shorter name I do not see a major problem.

Reply:
otelarrow sounds good to me.